api/agent/drivers/docker/docker.go

```diff
@@ -209,6 +274,7 @@ func (drv *DockerDriver) unpickNetwork(c *cookie) {
 
 func (drv *DockerDriver) CreateCookie(ctx context.Context, task drivers.ContainerTask) (drivers.Cookie, error) {
 
+	drv.imageCache.Lock(task.Image())
```

Comment:
Cookie close() needs Unlock(). And also recursive lock is needed as pointed out by @zootalures